src/parsers/manifest/dash/parse_adaptation_sets.ts

```diff
@@ -183,15 +184,30 @@ export default function parseAdaptationSets(
       const adaptationChildren = adaptation.children;
       const parsedAdaptations = acc.adaptations;
       const representationsIR = adaptation.children.representations;
+      const adaptationAvailabilityTimeOffset: number|undefined = (() => {
+        const { periodAvailabilityTimeOffset } = periodInfos;
+        if (adaptation.children.baseURL == null ||
+            adaptation.children.baseURL.attributes.availabilityTimeOffset == null) {
+          return periodAvailabilityTimeOffset;
+        }
+        if (periodAvailabilityTimeOffset != null) {
+          return adaptation.children.baseURL.attributes.availabilityTimeOffset +
+            periodAvailabilityTimeOffset;
+        }
+        return adaptation.children.baseURL.attributes.availabilityTimeOffset;
```

Comment:
What about setting `0` for the availabilityTimeOffset by default?

Like we did with the baseURL.

That way we can just add at each step the difference like we did with baseURL and string concatenation before